### ISSUE TITLE
Link Forms to Odoo 13 Objects

### ISSUE THREAD

**Issue by Majoul** [novacode-nl/odoo-formio#55]
Hello 

Odoo 13
I was able to create a form and publish it on the website. 
The form builder is based on Contacts (res.partner) 
I seem to miss something to make the links between the form fields /submission with the res.partner fields in order to add a new contact. 

Thank you for your help

**Comment by bobslee** [comment]
Hi,

Please can you share screenshots regarding the Form Builder configuration, tabs:

- Settings
- Components API

Did you installed the latest version (13.0.6.0) of the Formio module?

Best,
Bob

--
Vriendelijke groet / Best regards,

Bob Leers

Nova Code - Odoo software solutions
T: +31 (0)6 - 222 36 151
M: bob@novacode.nl
W: https://www.novacode.nl
________________________________
Van: Majoul <notifications@github.com>
Verzonden: Tuesday, September 1, 2020 11:45:53 AM
Aan: novacode-nl/odoo-formio <odoo-formio@noreply.github.com>
CC: Subscribed <subscribed@noreply.github.com>
Onderwerp: [novacode-nl/odoo-formio] Link Forms to Odoo 13 Objects (#55)


Hello

Odoo 13
I was able to create a form and publish it on the website.
The form builder is based on Contacts (res.partner)
I seem to miss something to make the links between the form fields /submission with the res.partner fields in order to add a new contact.

Thank you for your help

**Comment by bobslee** [comment]
By the way, it's not possible to link a new (public) Form with an existing Odoo object.
This only works with backend (non public) Forms, e.g. with modules: formio_sale, formio_partner, formio_crm.

With public (website Forms) following is possible:
- Link to Form to an existing partner (if found / matched by email address)
- If not found... then after Form submission one can create a new Partner with the Form linked.

**Comment by Majoul** [comment]
![settings](https://user-images.githubusercontent.com/10036464/91841482-c3857c00-ec52-11ea-8f4d-63e6ad2581de.png)
![component API](https://user-images.githubusercontent.com/10036464/91841484-c41e1280-ec52-11ea-9fee-ff25b8e5ecf1.png)


" 
> * Link to Form to an existing partner (if found / matched by email address)
> * If not found... then after Form submission one can create a new Partner with the Form linked."

This  really what I am missing 
Do I get it right that I can not add a new contact with a public form ?

**Comment by walkerblackwell** [comment]
I think in order to implement this you would need to make a python action that is triggered when the new form submission is made in the backend of odoo-formio. This python action would take the data from the submit and create a new contact. 

A note on security: Public contact creation would really need to require the recaptcha (so bots don’t fill the contacts) and also you would probably need to write some logic in the action on the backend to de-dupe or create sub-contacts if email is already a contact.

tldr: This formio is not an entire complete api for interacting with Odoo models but actions can be implemented after submission of data.

Best regards,
Walker

**Comment by bobslee** [comment]
Sir @walkerblackwell thanks for sharing your approach :+1: 

In a matter of fact, in version 5.3, I implemented a new feature with the configuration in the Form Builder, as shown in the screenshot.
I hope it wouldn't introduce some "feature creep" and is a nice feature to have in the base Formio module.

**Feature 5.3**
To create or link (existing) Partner with a Form submission, specify the fields (Email, Name) from the Components API / Property Names.
Partner determination/match shall be done by Email. This API is especially useful for public Forms. 

![image](https://user-images.githubusercontent.com/328803/91854810-9346d900-ec64-11ea-8cb0-49f97054f205.png)

CC: @Majoul

**Comment by bobslee** [comment]
@Majoul Configuring a "Resource Model" with a public Form is useless.
Implementation doesn't match a partner (contact) to link with. So please leave the "Resource Model" black / cleared.

Please use the configuration I explained in my previous comment OR follow the suggestions from @walkerblackwell.

Did it solved your question / request?

**Comment by bobslee** [comment]
Closed this issue, due to no response.
